pandas/tests/indexing/test_iloc.py

```diff
@@ -875,6 +875,13 @@ def test_iloc_setitem_dictionary_value(self):
         expected = DataFrame({"x": [1, 9], "y": [2.0, 99.0]})
         tm.assert_frame_equal(df, expected)
 
+    def test_iloc_truncate_data_wrong_axis_single_column(self):
+        # GH30263
+        df = DataFrame({"A": [1] * 10})
+        df = df.iloc[:, :5]
+        expected = np.array([10, 1])
+        tm.assert_equal(np.array(df.shape), expected)
```

Comment:
I think the issue only appears for ExtensionArray. I'd guess that the behavior you're testing for here is, in some shape or form tested for already, just not for ExtensionArray. Can you check?

Also, as reported in the original issue, `df.shape` returns the expected result but `df.values.shape` does not. Both should be tested (again, for EAs).